Firestore/core/src/firebase/firestore/api/settings.h

```diff
@@ -0,0 +1,88 @@
+
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_SETTINGS_H_
+#define FIRESTORE_CORE_SRC_FIREBASE_FIRESTORE_API_SETTINGS_H_
+
+#include <string>
+
+namespace firebase {
+namespace firestore {
+namespace api {
+
+/**
+ * Represents settings associated with a FirestoreClient.
+ *
+ * PORTING NOTE: We exclude the user callback std::executor in order to avoid
+ * ownership complexity.
+ */
+class Settings {
+ public:
+  Settings() = default;
+
+  void set_host(const std::string_view& value) {
+    host_ = value;
+  }
+  const std::string& host() const {
+    return host_;
+  }
+
+  void set_ssl_enabled(bool value) {
+    ssl_enabled_ = value;
+  }
+  bool ssl_enabled() const {
+    return ssl_enabled_;
+  }
+
+  void set_persistence_enabled(bool value) {
+    persistence_enabled_ = value;
+  }
+  bool persistence_enabled() const {
+    return persistence_enabled_;
+  }
+
+  void set_timestamps_in_snapshots_enabled(bool value) {
+    timestamps_in_snapshots_enabled_ = value;
+  }
+  bool timestamps_in_snapshots_enabled() const {
+    return timestamps_in_snapshots_enabled_;
+  }
+
+  void set_cache_size_bytes(int64_t value) {
+    cache_size_bytes_ = value;
+  }
+  int64_t cache_size_bytes() const {
+    return cache_size_bytes_;
+  }
+
+  friend bool operator==(const Settings& lhs, const Settings& rhs);
+
+  size_t Hash() const;
+
+ private:
+  std::string host_;
+  bool ssl_enabled_;
```

Comment:
Note: all the primitives will be uninitialized by the default constructor. I understand you initialize them right after creating a `Settings` object, but this still seems error-prone. I'd suggest using in-class initializers to prevent that (only for `bool`/`int64_t`, something like `bool ssl_enabled_ = false;`).

(Very optional: you could make `set` methods return a reference to `Settings` to enable chaining of the calls)

Reply:
Ooooh, good catch. I totally forgot that C++ thinks playing with uninitialized memory is a fun game.

I actually like the chaining idea, but I don't think it's probably worth re-working.  And actually, if I were to rework it I think I'd go back and forego having setters and go the giant constructor route since that avoids the uninitialized problem too...